base/errorshow.jl

```diff
@@ -372,16 +374,8 @@ function show_method_candidates(io::IO, ex::MethodError, @nospecialize kwargs=()
             end
             s1 = sig0.parameters[1]
             sig = sig0.parameters[2:end]
-            print(iob, "  ")
-            if !isa(func, rewrap_unionall(s1, method.sig))
-                # function itself doesn't match
-                continue
-            else
-                # TODO: use the methodshow logic here
-                use_constructor_syntax = isa(func, Type)
-                print(iob, use_constructor_syntax ? func : typeof(func).name.mt.name)
-            end
-            print(iob, "(")
+            print(iob, "  ", func, "(")
```

Comment:
Something like this to still highlight mismatches, like other arguments?
```suggestion
print(iob, "  ")
if !isa(func, rewrap_unionall(s1, method.sig))
    ... Base.with_output_color(...); print(iob, func); end ...
else
    print(iob, func)
end
print(iob, "("))
```

Reply:
IMO that will be a very bad effort/result ratio, it the `...` are extended. I would prefer to leave it as is.